academic-services/knowledge-exploration-service/tutorial-grammar-design.md

```diff
@@ -0,0 +1,392 @@
+
+| Attribute Name | Description| Index Data Type | Index Operations |
+| ---- | ---- | ---- | ---- |
+| `Abstract.Words` | A words array containing all distinct words from the publication's abstract. Used for supporting partial matches in abstract search. | `string*` | `["equals"]` |
+| `DOI` | Indicates that "DOI" attribute is a object composed of multiple attributes | `Composite?` | - |
+| `DOI.Name` | Used for providing attibute match in DOI search. | `string?` | `["equals"]` |
+| `Title` | Indicates that "Title" attribute is an object composed of multiple attributes. | `Composite?` | - |
+| `Title.Name` | Used for providing attribute match in title search. | `blob?` | - |
+| `Title.Words` | A words array containing all distinct words from the publication's title. Used for supporting partial matches in title search. | `string*` | `["equals"]` |
+| `FullTextUrl` | Indicates that "FullTextUrl" attribute is an object composed of multiple attributes | `Composite?` | - |
+| `FullTextUrl.Name` | Used for providing direct attribute match in FullTextUrl search| `string?` | `["equals"]` |
+
+#### Search attributes
+
+Attributes that we plan to use for processing search queries need to be indexed, similar to filter attributes in the ["Build a library browser with contextual filters"](tutorial-schema-design.md) tutorial. For example, to enable our application to process search queries like "papers from microsoft about machine learning" we need to enable the `equals` index operation on the `FieldsOfStudy.Name` and `AuthorAffiliations.AffiliationName` attributes.
+
+Before indexing the attributes, we also want to normalize the attributes to improve search results. For example, we can improve publication title search accuracy by normalizing both the publication titles and the search queries to lowercase characters only. This ensures that publication title quries that have case mismatches will still yield search results. For more normalization details, see `MakesInteractor.NormalizeStr(queryStr)` in `makesInteractor.js`
+
+To enable fuzzy search or keyword search, we may also transform the attributes before indexing the attributes. For example, we enable abstract keywords search by transforming the abstract of a publication `Abstract.OriginalName` into `Abstract.Words` such that unique words from the abstract can be indexed.
+
+## Design a search grammar
+
+Now that we have the appropriate data fields in our index, we will design a grammar tailored to process natural language search queries. In this tutorial, we will use the [sample grammar](samplePrivateLibraryData.linked.search.grammar.xml) to illustrate how to support **direct attribute search**, **multiple attribute search**, **partial attribute search**, **semantic search**, and **drop terms with penalties**.
```

Comment:
>semantic search [](start = 361, length = 15)

Isn't this redundant? Everything you've mentioned before this *is* semantic search